app/cli/internal/action/integration_list.go

```diff
@@ -46,28 +51,56 @@ func (action *IntegrationList) Run() ([]*IntegrationItem, error) {
 
 	result := make([]*IntegrationItem, 0, len(resp.Result))
 	for _, p := range resp.Result {
-		result = append(result, pbIntegrationItemToAction(p))
+		i, err := pbIntegrationItemToAction(p)
+		if err != nil {
+			return nil, err
+		}
+
+		result = append(result, i)
 	}
 
 	return result, nil
 }
 
-func pbIntegrationItemToAction(in *pb.IntegrationItem) *IntegrationItem {
+func pbIntegrationItemToAction(in *pb.IntegrationItem) (*IntegrationItem, error) {
 	if in == nil {
-		return nil
+		return nil, errors.New("nil input")
 	}
 
 	i := &IntegrationItem{
 		Kind: in.GetKind(), ID: in.GetId(),
 		CreatedAt: toTimePtr(in.GetCreatedAt().AsTime()),
 	}
 
-	if c := in.GetConfig().GetDependencyTrack(); c != nil {
```

Comment:
This is the key of this change. Instead of being explicit about a specific type we take a generic type (anypb.Any) and create the map from it. 

In other words, it makes this client code (and the attachment one) generic for any integration